evergreen.yml

```diff
@@ -1227,7 +1229,7 @@ functions:
 
           ~/.cargo/bin/rustup default nightly
           ~/.cargo/bin/rustup target add x86_64-unknown-linux-gnu
-          export RUSTFLAGS="-Z sanitizer=address"
+          export RUSTFLAGS="-Z sanitizer=leak"
```

Comment:
I think it is important to assess that we don't overflow buffer or do out of bound access just as much as making sure that we don't have memory leaks.
I thought that since leak is already on by default on Linux, setting the flag to "address" would mean that we cover both, but looking at the rust doc, it looks like only one sanitizer at a time is possible.
Sorry to be a pain, but could we have 2 tasks?
- Run-time memory leak detection (aka run result set test with sanitizer=leak)
- Memory error detection (only need to compile the driver if I understand correctly)
To be fair, we might even need 3 with memory for even better coverage but this is outside of the scope for this and I think we can cover this better when dealing with SSDLC

Reply:
Currently, both ASAN functions (result set and integration) are commented out because they only test for leaks in the tests. Are you saying that we uncomment those and then add another task that just does memory error detection?

Reply:
Leak is a runtime detection. It should be used in conjunction with the unit tests.
For Address it is a compile time detection, so we just need to build the code with the flag.
So, I think we want to change the sanitizier to leak on "run asan unit tests" and have a new task in "Static code analysis" which build the driver with sanitizer=address.
Doe that make more sense?

Reply:
Yes sounds good!

Reply:
Okay, if I understand correctly we want to run the unit test with `sanitizer=leak` and add a task that builds the driver with `sanitizer=address`.  I added `asan-compile` and it runs cargo build with `sanitizer=address` with the debug build.  I can add --release if we think it will help.  I was able to get it to compile without the error I mentioned in the other thread by adding `--target x86_64-unknown-linux-gnu` to the build command.

Reply:
Yes, you need `--target x86_64-unknown-linux-gnu` just a funny quirk of Rust. I think it's because it's still a nigthtly feature